### PR TITLE
Adding function which overlays rectangles onto an image.

### PR DESCRIPTION
**Pull request by zivy**
Overlay bounding boxes onto image. Supports various bounding box specifications which are commonly used in object detection.

### CODE REVIEW

**Comment by blowekamp** [comment]
This seems like a good utility method. Thanks for working on it.

**Comment by zivy** [comment]
@blowekamp, please approve so can merge this, thanks.

**Review by blowekamp — APPROVED** [review]

File: SimpleITK/utilities/overlay_bounding_boxes.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    bounding box:
+    "MINXY_MAXXY" - [min_x, min_y, max_x, max_y]
+    "MINXY_WH" - [min_x, min_y, width, height]
+    "CENT_WH" - [center_x, center_y, width, height]
+    "CENT_HALFWH" - [center_x, center_y, width/2, height/2]
+    Bounding boxes are plotted in the order they appear in the iterable/list. To change the overlap between rectangles
+    change the order in the list. The last entry in the list will be plotted on top of the previous ones.
+
+    Caveat: When using larger line widths, bounding boxes that are very close to the image border may cause an exception
+            and result in partial overlay. A trivial solution is to decrease the value of the half_line_width parameter.
```
[the remark this answers is not in the document]
Comment (zivy) [reply SimpleITK/utilities/overlay_bounding_boxes.py:42]: Nope, this is an input error, just plot and return a boolean indicating that one or more bounding boxes were outside the image.

File: SimpleITK/utilities/overlay_bounding_boxes.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            ]
+            for bbox in original
+        ],
+        "CENT_HALFWH": lambda original: [
+            [bbox[0] - bbox[2], bbox[1] - bbox[3], bbox[0] + bbox[2], bbox[1] + bbox[3]]
+            for bbox in original
+        ],
+    }
+
+    # Confirm image is in expected format
```
[the remark this answers is not in the document]
Comment (zivy) [reply SimpleITK/utilities/overlay_bounding_boxes.py:83]: Documentation updated to reflect that only 2D images.

File: SimpleITK/utilities/overlay_bounding_boxes.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+import SimpleITK as sitk
+from typing import Sequence
+
+
+def overlay_bounding_boxes(
+    image: sitk.Image,
+    bounding_boxes: Sequence[Sequence[float]],
+    bounding_box_format: str = "MINXY_MAXXY",
+    normalized: bool = False,
+    colors: Sequence[Sequence[int]] = [],
```
[the remark this answers is not in the document]
Comment (zivy) [reply SimpleITK/utilities/overlay_bounding_boxes.py]: Color now specified using a flat Sequence.

File: SimpleITK/utilities/overlay_bounding_boxes.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+            ] = vert
+            overlay_image[
+                bbox[0] : bbox[2],  # noqa E203
+                bbox[1] - half_line_width : bbox[1] + half_line_width + 1,  # noqa E203
+            ] = horiz
+            overlay_image[
+                bbox[0] : bbox[2],  # noqa E203
+                bbox[3] - half_line_width : bbox[3] + half_line_width + 1,  # noqa E203
+            ] = horiz
+        except Exception:  # Drawing outside the border of the image will cause problems
```
[the remark this answers is not in the document]
Comment (zivy) [reply SimpleITK/utilities/overlay_bounding_boxes.py:169]: Changed return type to include an error value if oner or more bounding boxes were outside the image.

File: SimpleITK/utilities/overlay_bounding_boxes.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    colors: Sequence[Sequence[int]] = [],
+    half_line_width: int = 0,
+) -> sitk.Image:
+    """
+    Overlay axis aligned bounding boxes on an image. The function supports several ways of specifying a
+    bounding box:
+    "MINXY_MAXXY" - [min_x, min_y, max_x, max_y]
+    "MINXY_WH" - [min_x, min_y, width, height]
+    "CENT_WH" - [center_x, center_y, width, height]
+    "CENT_HALFWH" - [center_x, center_y, width/2, height/2]
```
[the remark this answers is not in the document]
Comment (zivy) [reply SimpleITK/utilities/overlay_bounding_boxes.py:37]: Index, documentation updated to reflect this.

**Pull request merged**